src/ast/passes/attachpoint_passes.cpp

```diff
@@ -374,8 +374,14 @@ int AttachPointParser::parse()
     probe->attach_points.erase(it.begin(), it.end());
 
     if (probe->attach_points.empty()) {
-      probe->addError() << "No attach points for probe";
-      failed++;
+      const auto missing_probes = bpftrace_.config_->missing_probes;
+      std::string msg = "No attach points for probe";
+      if (missing_probes == ConfigMissingProbes::error) {
+        probe->addError() << msg;
+        failed++;
+      } else if (missing_probes == ConfigMissingProbes::warn) {
+        probe->addWarning() << msg;
```

Comment:
I don't think this warning message is needed either.

And actually, this `parse` function can be cleaned up as it doesn't need to return an `int` as nothing is consuming it.